src/components/VideoPreview.tsx

```diff
@@ -0,0 +1,55 @@
+import { useVideo } from '@/api/hooks'
+import routes from '@/config/routes'
+import React from 'react'
+import VideoPreviewBase from '../shared/components/VideoPreviewBase/VideoPreviewBase'
+
+type VideoPreviewProps = {
+  id?: string
+  showChannel?: boolean
```

Comment:
Maybe to make this a little less coupled with the `VideoPreviewBaseProps`, we could extract those "meta" props to a separate type? That type would contain the `showChannel`, `showMeta`, `main` and the event handlers. With that we can say that `type VideoPreviewProps = { ... } & VideoPreviewBaseMetaProps`. Then in the component we also can do `({ id, ..., ...metaProps })` and pass `metaProps` directly to `VideoPreviewBase`

Reply:
Yeah I think this is much better, the repetition was bothering me a bit, then there is the progress prop that is an inbetween